internal/model/image.go

```diff
@@ -16,6 +20,32 @@ type Image struct {
 	HubTpl      string            `yaml:"hub_tpl,omitempty" json:",omitempty"`
 	HubLink     string            `yaml:"hub_link,omitempty" json:",omitempty"`
 	Metadata    map[string]string `yaml:"metadata,omitempty" json:",omitempty"`
+
+	watchRepoWasSet bool
+}
+
+func (i *Image) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var tmpStruct map[string]interface{}
+	if err := unmarshal(&tmpStruct); err != nil {
+		return fmt.Errorf("failed unmarshalling into tmp struct: %w", err)
+	}
+
+	type imageAlias Image
+	tmpImage := imageAlias{}
+
+	if err := unmarshal(&tmpImage); err != nil {
+		return err
+	}
+
+	*i = Image(tmpImage)
+
+	_, i.watchRepoWasSet = tmpStruct["watch_repo"]
+
+	return nil
+}
+
+func (i Image) WatchRepoWasSet() bool {
+	return i.watchRepoWasSet
```

Comment:
Don't think this is needed. If you want to check `WatchRepo` has been set or not, its type should be changed to a pointer. Pretty much like we do for other structs such as https://github.com/crazy-max/diun/blob/fa7f290f05d5810e01099937e6eb955198f28253/internal/model/notif_discord.go#L13

Reply:
I went down this route initially and backed out. Testing became more complicated with this approach as you can’t set a pointer literal boolean value. Eg. `&true` doesn’t work. Workaround seem to include `&[]bool{true}[0]`, which is ads some complexity. I can give it another go. I think it really only matters for testing, so maybe I can do `const True = true` and then reference `&True` or something.

Reply:
Oh, looks like `pkg/utl/utl.go` has `NewTrue()` and `NewFalse()` to work around this issue. I’ll refactor this part to use those.